gitinfo.go

```diff
@@ -37,8 +38,18 @@ func collectGitInfo(ref string) *Git {
 	results := map[string]string{}
 	gitPath, err := exec.LookPath("git")
 	if err != nil {
-		log.Fatal(err)
+		log.Printf("fail to look path of git: %v", err)
+		log.Print("git information is omitted")
+		return nil
 	}
+
+	if ref != "HEAD" {
+		// make sure that the commit is in the local
+		// e.g. shallow cloned repository
+		_, _ = runCommand(gitPath, "fetch", "--depth=1", "origin", ref)
```

Comment:
Does this fix the issue with `checkout@v2`?

Reply:
yes, it does
see `.github/workflows/main.yml` for more details